src/LocalParcel.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Amp\Sync;
+
+/**
+ * @template T
+ * @template-implements Parcel<T>
+ */
+final class LocalParcel implements Parcel
+{
+    /**
+     * @param Semaphore $semaphore
+     * @param T $value
+     */
+    public function __construct(
+        private Semaphore $semaphore,
```

Comment:
It should require a `Mutex` instead, no?

Reply:
We allow a `Semaphore` on `Amp\Sync\synchronized()`. I would generally think both should be a `Mutex` actually.

Reply:
Yeah, `synchronized` should be a `Mutex`, too.